### ISSUE TITLE
keyboard focus for foot does not work  with mouse click right after unlock hyprlock

### ISSUE THREAD

**Issue by rtgiskard** [hyprwm/Hyprland#5884]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit cba1ade848feac44b2eda677503900639581c3f4  (props: bump version to 0.40.0).
Date: Sat May 4 23:42:32 2024
Tag: v0.40.0, commits: 4606

flags: (if any)


System Information:
System name: Linux
Node name: rt
Release: 6.8.3-gskd
Version: #1 SMP PREEMPT_DYNAMIC Thu, 04 Apr 2024 10:17:48 +0000


GPU information: 
c7:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Phoenix1 [1002:15bf] (rev c2) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```

</details>

### Bug or Regression?

Bug

### Description

I'm using hyprlock to lock the screen, if there be a firefox and a foot window before the lock, once unlocked, click on foot window does not activate keyboard focus for foot window, but click on firefox works as expected.

### How to reproduce

1. open firefox and foor side by side, then lock the screen with hyprlock
2. unlock
3. click foot, then input chars

. you can't actually input chars into foot terminal in step 3
. if you click firefox in step 3, everything works well
. if you open another window or click firefox then click foot, foot will get keyboard focus
. mouse focus works as expected all the time

**Comment by jasinco** [comment]
I met some problems seem like this one, I switch window focus between firefox and kitty, and sometimes I can't type any words on kitty's window. Hyprland render the colorful border on kitty, but doesn't take any input.

**Comment by MorrowSol** [comment]
I have the same issue. When switching focus from Firefox to Kitty using the mouse, Kitty displays the focus color, but the keyboard focus remains on Firefox. I tried downgrading Hyprland from 0.39.1 to 0.39.0, but it didn't work. Not only these two programs, but also all other software may exhibit the same issue.

**Comment by rtgiskard** [comment]
@jasinco @MorrowSol the issue you referred to is likely to be this one: https://github.com/hyprwm/Hyprland/issues/5815, which should have been resolved with hyprland v0.40.0 (use hyprland-git)

this is another issue related to keyboard focus on the first window after unlock

**Comment by LewentalDigital** [comment]
I have the same issue

**Comment by esatgundogdu** [comment]
I have the same issue too, any solution?

**Comment by rtgiskard** [comment]
I'd like to resolve the issue, but have no concept for the through handle of focus like keyboard and mouse 😵, is there any related doc for that?

**Comment by vaxerski** [comment]
code is all you get pretty much. See `CInputManager::mouseMoveUnified` to start. Keyboard focus is spread across many places (keybinds, etc) but `CInputManager::mouseDownNormal` and the aforementioned should be a good start

**Comment by MightyPlaza** [comment]
unlikely to help, but this issue has been present for a long time (at least v0.35) and happens on sway-lock as well

this seems like a similar case of a weird state that happens somehow in multiple places, where the hyprland thinks the window is focused, but it does not have keyboard focus
related:https://github.com/hyprwm/Hyprland/issues/4848 and https://github.com/hyprwm/Hyprland/issues/4973

**Comment by LewentalDigital** [comment]
I have noticed that this issue also happens with wofi and swaync, but only in specific conditions:
I have a waybar module to show/hide wofi menu on click and another to show/hide swaync notification center on click, if I start wofi or swaync and then **close them by clicking** on their waybar modules, mouse action is possible but keyboard does not focus on programs under mouse cursor.

This situation does not happen if i close wofi or swaync using **key binding** set in hyprland conf.

Maybe this is linked to hyprlock or can help understand the issue.
